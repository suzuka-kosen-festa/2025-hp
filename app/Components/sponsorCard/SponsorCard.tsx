/** @jsxImportSource @emotion/react */
import type { ReactNode } from "react";
import Box from "@mui/material/Box";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import Stack from "@mui/material/Stack";
import { useTheme } from "@mui/material/styles";
import Typography from "@mui/material/Typography";
import useMediaQuery from "@mui/material/useMediaQuery";
import * as React from "react";
import { ja } from "../../locales/ja";

export interface SponsorCardComponentProps {
  image: string
  sponsorName: string
  description: string
  phone?: string // ä¾‹: "059-389-7860"
}

function SponsorCard(props: SponsorCardComponentProps): ReactNode {
  const { image, sponsorName, description, phone } = props;
  const theme = useTheme();
  const isLgUp = useMediaQuery(theme.breakpoints.up("lg"));

  const handleImageError = (e: React.SyntheticEvent<HTMLImageElement, Event>) => {
    if (e.currentTarget.src.endsWith("/images/not-found.png"))
      return;
    e.currentTarget.src = "/images/not-found.png";
  };

  return (
    <Card
      elevation={4}
      sx={{
        mx: "auto",
        maxWidth: { xs: 360, sm: 560, lg: 980 },
        borderRadius: { xs: 2, sm: 3 },
        px: { xs: 1, sm: 0 },
        py: { xs: 1, sm: 0 },
      }}
    >
      <Stack
        direction={{ xs: "column", lg: "row" }}
        sx={{ alignItems: "stretch", gap: { xs: 1, sm: 0 } }}
      >
        <Box
          sx={{
            flexShrink: 0,
            width: { xs: "100%", lg: 420 },
            position: "relative",
          }}
        >
          <CardMedia
            component="img"
            image={image}
            alt={ja.sponsorCard.imageAlt(sponsorName)}
            onError={handleImageError}
          />
        </Box>
        <CardContent
          sx={{
            display: "flex",
            flexDirection: "column",
            justifyContent: "center",
            gap: 1.5,
            p: { xs: 2, sm: 3 },
          }}
        >
          <Typography
            variant={isLgUp ? "h5" : "h6"}
            component="h2"
            sx={{ fontWeight: 700, textAlign: { xs: "center", lg: "left" }, fontSize: { xs: "1.1rem", sm: "1.25rem", lg: "inherit" } }}
          >
            {ja.sponsorCard.name(sponsorName)}
          </Typography>

          <Typography
            variant="body1"
            sx={{ color: "text.secondary", whiteSpace: "pre-line", fontSize: { xs: "0.9rem", sm: "1rem" }, lineHeight: { xs: 1.5, sm: 1.7 } }}
          >
            {description}
          </Typography>

          {phone && (
            <Box sx={{ display: "flex", alignItems: "center", gap: 1, mt: 1 }}>
              <Typography
                variant="body1"
                component="a"
                href={`tel:${phone.replace(/-/g, "")}`}
                sx={{ textDecoration: "none", color: "inherit", fontSize: { xs: "0.95rem", sm: "1rem" } }}
              >
                {phone}
              </Typography>
            </Box>
          )}
        </CardContent>
      </Stack>
    </Card>
  );
}

export default SponsorCard;
