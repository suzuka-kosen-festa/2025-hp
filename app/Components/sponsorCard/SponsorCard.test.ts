// @vitest-environment jsdom

import { composeStories } from "@storybook/react";
import { expect, it } from "vitest";
import * as stories from "./SponsorCard.stories";


const { Default } = composeStories(stories);

it("sponsor card snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
