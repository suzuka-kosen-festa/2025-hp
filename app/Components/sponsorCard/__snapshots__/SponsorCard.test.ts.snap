// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sponsor card snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiCard-root css-1rpa8bx-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiStack-root css-zs2mjy-MuiStack-root"
    >
      <div
        class="MuiBox-root css-19o3cmm"
      >
        <img
          alt="テストスポンサーのイメージ"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1kuwa16-MuiCardMedia-root"
          src="https://placehold.jp/800x400.png"
        />
      </div>
      <div
        class="MuiCardContent-root css-4p6oay-MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 css-fvf604-MuiTypography-root"
        >
          テストスポンサー
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1rwwn4c-MuiTypography-root"
        >
          スポンサーの説明がここに入ります。スポンサーの説明がここに入ります。スポンサーの説明がここに入ります。スポンサーの説明がここに入ります。
        </p>
        <div
          class="MuiBox-root css-1mpg011"
        >
          <a
            class="MuiTypography-root MuiTypography-body1 css-411ob9-MuiTypography-root"
            href="tel:xxxxxxxxxxx"
          >
            xxx-xxxx-xxxx
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
